### PR TITLE
Refactor event tracker out of asio class

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This refactors the event tracker to be decoupled from the asio class.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/common/event_stats.h [hunk 1]
```diff
@@ -0,0 +1,170 @@
+// Copyright 2017 The Ray Authors.
```
Comment (ericl) [inline src/ray/common/event_stats.h:1]: This file moved verbatim except for renaming.

File: src/ray/common/event_stats.cc [hunk 2]
```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Ray Authors.
```
Comment (ericl) [inline src/ray/common/event_stats.cc:1]: This file moved verbatim except for renaming.

**Review by ericl — APPROVED** [review]

File: src/ray/common/asio/instrumented_io_context.h [hunk 3]
```diff
@@ -121,79 +51,13 @@ class instrumented_io_context : public boost::asio::io_context {
...
+  void dispatch(std::function<void()> handler, const std::string name)
       LOCKS_EXCLUDED(mutex_);
 
-  /// Builds and returns a statistics summary string. Used by the DebugString() of
-  /// objects that used this io_context wrapper, such as the raylet and the core worker.
-  ///
-  /// \return A stats summary string, suitable for inclusion in an object's
-  /// DebugString().
-  std::string StatsString() const LOCKS_EXCLUDED(mutex_);
+  EventTracker &stats() const { return *event_stats_; };
```
Comment (clarkzinzow) [inline src/ray/common/asio/instrumented_io_context.h:58]: Why bypass the shared pointer reference counting with the dereference + reference?

**Review by clarkzinzow — APPROVED** [review]
LGTM!

File: src/ray/common/asio/instrumented_io_context.h (hunk 3 above)
Comment (rkooo567) [reply src/ray/common/asio/instrumented_io_context.h:58]: In this case, we can make event_stats a unique pointer?

Comment (ericl) [reply src/ray/common/asio/instrumented_io_context.h:58]: This should only be used for non-owning access. The reason we have event_stats as a shared_ptr is so we can pull it out of the asio class (TODO later on).

**Pull request merged**